### ISSUE TITLE
Import classes from the same package

### ISSUE THREAD

**Issue by Emmurillo** [swagger-api/swagger-codegen#2141]
Would be nice to avoid unused imports by checking if the imported class is part of the same package.

**Comment by wing328** [comment]
For which language?

**Comment by Emmurillo** [comment]
Java

**Comment by wing328** [comment]
Do you mind showing your generated code as an example to illustrate the issue/enhancement?

**Comment by wing328** [comment]
If you've time, please feel free to submit a PR to improve the package import in Java.

Ref: https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md

**Comment by Emmurillo** [comment]
There you go:

![screenshot from 2016-02-19 10 22 52](https://cloud.githubusercontent.com/assets/5572221/13181477/f645f022-d6f2-11e5-954e-085e44d31b93.png)

That's the kind of import that could be improved.

**Comment by Clijsters** [comment]
Is there any progress on this? I'm facing the same problem, over 1y later. Codegen just adds imports to every used class, even if it's in the same package.

**Comment by ranimg** [comment]
Yes, version swagger-codegen version used is 2.3.1,  facing in java and also Python. Python breaks if a class is imported as itself.
